openedx/xblock-skill-tagging
# feat: replace signal handler with event producer config

**Description:** Replace the hard-coded signal handler with event bus producer config. See https://github.com/openedx/openedx-events/pull/249 and https://github.com/openedx/edx-platform/pull/32874 for more details.

**JIRA:** `Private-ref`: [BB-7666](https://tasks.opencraft.com/browse/BB-7666)

**Dependencies:** 

* https://github.com/openedx/openedx-events/pull/249
* https://github.com/openedx/edx-platform/pull/32874

**Merge deadline:** List merge deadline (if any)

**Testing instructions:**

* Setup master devstack
* Run `make lms-up` to get lms up and running.
* Create lms shell using `make lms-shell`.
* Install this plugin by running `pip install -e /edx/src/xblock-skill-tagging/` (assuming this repo is cloned in `<devstack-base-dir>/src/`)
* Include `EVENT_BUS_PRODUCER_CONFIG` setting as per the configuration in `README` in `lms/envs/private.py` to push `XBLOCK_SKILL_VERIFIED` to event bus.
* Run below snippet to send `XBLOCK_SKILL_VERIFIED` event.
```python
from datetime import datetime, timezone
from openedx_events.learning.signals import XBLOCK_SKILL_VERIFIED
from openedx_events.learning.data import XBlockSkillVerificationData
XBLOCK_SKILL_VERIFIED.send_event(
    time=datetime.now(timezone.utc),
    xblock_info=XBlockSkillVerificationData(
        usage_key="block-v1:edx+DemoX+Demo_course+type@video+block@UaEBjyMjcLW65gaTXggB93WmvoxGAJa0JeHRrDThk",
        verified_skills=[1, 2],
        ignored_skills=[3, 4],
    )
)
```
* It should print something like below:
```log
2023-08-29 12:03:46,477 INFO 1502 [edx_event_bus_redis.internal.producer] [user None] [ip None] producer.py:82 - Message delivered to Redis event bus: topic=dev-learning-custom-xblock-skill-verfied, message_id=1b5934e4-4664-11ee-83e5-0242ac12000b, signal=<
OpenEdxPublicSignal: org.openedx.learning.xblock.skill.verified.v1>, redis_msg_id=b'1693310626477-0'
[(<function general_signal_handler at 0x7fcf6e0e2a60>, None)]
```
* You can play with the topic name and other setting to see the changes.


**Reviewers:**
- [ ] @sameenfatima78 

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
